__tests__/app.test.js

```diff
@@ -312,3 +313,125 @@ describe('POST /api/articles/:article_id/comments', () => {
+        const articleId = 1;
+        const patch = {inc_votes: 100};
+        const expectedOutput = {...data.articleData[articleId - 1]};
+        expectedOutput.votes += patch.inc_votes;
+        // adjust for hour's difference between dataset and database
+        const createDate = new Date(expectedOutput.created_at);
+        expectedOutput.created_at = new Date(createDate.setHours(createDate.getHours() - 1)).toISOString();
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(201)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Array.isArray(body.article)).not.toBe(true);
+                expect(body.article).toMatchObject(expectedOutput);
+                expect(body.article.votes).toBe(expectedOutput.votes);
+            });
+    });
+    test('201: should return an updated patched article with a decrease in votes', () => {
+        const articleId = 1;
+        const patch = {inc_votes: -50};
+        const expectedOutput = {...data.articleData[articleId - 1]};
+        expectedOutput.votes += patch.inc_votes;
+        // adjust for hour's difference between dataset and database
+        const createDate = new Date(expectedOutput.created_at);
+        expectedOutput.created_at = new Date(createDate.setHours(createDate.getHours() - 1)).toISOString();
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(201)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Array.isArray(body.article)).not.toBe(true);
+                expect(body.article).toMatchObject(expectedOutput);
+                expect(body.article.votes).toBe(expectedOutput.votes);
+            });
+    });
+    test('201: should return an unchanged article when votes not changed', () => {
+        const articleId = 1;
+        const patch = {inc_votes: 0};
+        const expectedOutput = {...data.articleData[articleId - 1]};
+        expectedOutput.votes += patch.inc_votes;
+        // adjust for hour's difference between dataset and database
+        const createDate = new Date(expectedOutput.created_at);
+        expectedOutput.created_at = new Date(createDate.setHours(createDate.getHours() - 1)).toISOString();
+        return request(app)
+            .patch(`/api/articles/${articleId}`)
+            .send(patch)
+            .expect(201)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Array.isArray(body.article)).not.toBe(true);
+                expect(body.article).toMatchObject(expectedOutput);
+                expect(body.article.votes).toBe(expectedOutput.votes);
+            });
+    });
+    test('201: should return an updated patched article with an increase in votes if supplied as a string', () => {
```

Comment:
This is interesting, and i can see why you have done the regex to handle this case, however, ideally we want to rely on a PSQL error for if we are given a string rather than number. 

When we get to front end, we can for instance parse the inc_votes value if a string and see whether it is a number or not.